### ISSUE TITLE
react-redux/no-unused-prop-types: False positive on props used within react-dnd dropSpec

### ISSUE THREAD

**Issue by bhollis** [yannickcr/eslint-plugin-react#3058]
Since upgrading to 7.25.0 I get false positives on prop types defined in TypeScript that are used within react-dnd dropSpec functions but not within the component itself. e.g. https://github.com/DestinyItemManager/DIM/blob/master/src/app/destiny1/loadout-builder/LoadoutBuilderDropTarget.tsx

**Comment by ljharb** [comment]
tests/lib/rules/no-this-in-sfc.js

**Comment by traverse** [comment]
Running into a similar issue reproducible with the following code

```tsx
interface Props {
  propA: string
  propB: number // marked as unused
}

function MyComponent(props: Props) {
  someLogicThatUsesProps(props)
  
  return <div>{props.propA}</div>
}

function someLogicThatUsesProps(props: Props) {
  if (props.propA !== '') return props.propA
  if (props.propB !== 0) return props.propB

  return 'some default'
}
```
In my case there's a bit more logic involved and several more props but I think it gets the point across.

**Comment by ljharb** [comment]
@traverse in that case, that's just the limits of static analysis - you should be destructuring `props` and passing things explicitly - props objects should never be directly passed anywhere else, for a number of reasons.

**Comment by bhollis** [comment]
This is *not* fixed in 7.25.1.

**Comment by bhollis** [comment]
Sorry, I didn't catch that this is in eslint-plugin-react-redux, not eslint-plugin-react.
